internal/collector/mail_delivery.go

```diff
@@ -0,0 +1,110 @@
+/*******************************************************************************
+*
+* Copyright 2025 SAP SE
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You should have received a copy of the License along with this
+* program. If not, you may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*
+*******************************************************************************/
+
+package collector
+
+import (
+	"context"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/sapcc/go-bits/jobloop"
+	"github.com/sapcc/go-bits/sqlext"
+
+	"github.com/sapcc/limes/internal/db"
+)
+
+type MailRequest struct {
+	ProjectID db.ProjectID `json:"project_id"`
+	Subject   string       `json:"subject"`
+	MimeType  string       `json:"mime_type"`
+	MailText  string       `json:"mail_text"`
+}
+
+const (
+	// how long to wait after error before retrying sending mails
+	mailDeliveryErrorInterval = 24 * time.Hour
+)
+
+func (c *Collector) MailDeliveryJob(registerer prometheus.Registerer, client MailDelivery) jobloop.Job {
+	return (&jobloop.ProducerConsumerJob[MailDeliveryTask]{
+		Metadata: jobloop.JobMetadata{
+			ReadableName: "mail delivery",
+			CounterOpts: prometheus.CounterOpts{
+				Name: "limes_mail_delivery",
```

Comment:
Names for counter metric families should be plural:
```suggestion
				Name: "limes_mail_deliveries",
```